documentation/projects/proposals/rekognition_data/20240510-implementation_plan_machine_generated_tags_frontend.md

```diff
@@ -0,0 +1,196 @@
+two subsections in the Generated tags section: tags generated by `recognition`
+and tags generated by `clarifai`. In the future, if there are more providers,
+there could be more tag sections. The tags in each subsection will be ordered by
+their accuracy.
+
+There is no need to create a separate component for machine-generated tags, the
+slot in the `VTag` component will enable us to display the additional accuracy
+information.
+
+### Add "About tags" description page
+
+We should add a page to explain the users the difference between where the
+Openverse tags come from, the difference between the source and generated tags,
+and a short description of the machine-generated tag providers. This page should
+be linked from the Generated tags section.
+
+## Dependencies
+
+This project needs the API to add the `unstable__provider` property to tags in
+media responses, which was implemented in
+[Expose provider in the API tags response #4280](https://github.com/WordPress/openverse/pull/4280)
+
+### Feature flags
+
+<!-- List feature flags/environment variables that will be utilised in the development of this plan. -->
+
+The project will contain at most 2 PRs that can be reverted, so the feature
+flags are not necessary.
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+No infrastructure changes will be necessary, we will only update the single
+result pages of the Nuxt frontend.
+
+## Alternatives
+
+<!-- Describe any alternatives considered and why they were not chosen or recommended. -->
+
+We have the provider (currently, it will be "clarifai" and "rekognition") and
+the accuracy data for the machine-generated tags.
+
+There are several options of displaying these properties.
+
+1. Do not display this data. The two tags sections will be identical.
+2. The generated tags will have a tooltip that displays the tag's provider and
+   its accuracy.
+3. Display generated tags in sections by provider so that we don't duplicate the
+   tag provider on each tag. Then, on each tag, display the accuracy.
+
+The first option is the simplest to implement, however, it fails to provide the
+fullest information to the users. The second option provides the accuracy and
+provider information to the user, however it is not very accessible. Different
+browsers and screen readers treat the tooltips differently. We have enough room
+on the page to display this data in full, using the third option: labelling all
+the tags generated by a single provider with a "by provider" heading, and adding
+the accuracy to the tag's inner content. This way, the user will see the
+provider and the accuracy of the tag without any additional actions. As a bonus,
+if a tag is generated by multiple providers, the user will see that different
+providers generate the same tag, and will be able to see the difference in
+accuracy between the providers.
```

Comment:
I'm drawn to go with option 1.

> …it fails to provide the fullest information to the users.

This part assumes that users need this information in the media details page for a better search experience. Since we don't have users data about how they relate to machine-generated information, I would go with the simplest solution.

Providing all information is certainly necessary, but it needs to be displayed in context to not load with unnecessary information that risks confusing visitors. Describing how tags are generated in an internal page meets this need as it allows more technical description.

Reply:
So, you think that showing that the tag is machine-generated or created by the source is enough, @fcoveram, and additional information for each tag would be confusing? All that we will display now on the single result page is the tag name.

Reply:
> So, you think that showing that the tag is machine-generated or created by the source is enough, and additional information for each tag would be confusing?

Yes to both questions.

> All that we will display now on the single result page is the tag name.

Perhaps next to the "Generated tags" sentence, we can include a link saying "how does this work?" (TBD) that takes you to a "machine-generated content" section in the About page. Here is a quick mockup.

<img src="https://github.com/WordPress/openverse/assets/895819/0f7a3d52-a81b-41ba-b273-b448e189e1e1" alt="media details page. generated tags section" width=300 >

Reply:
I agree, great discussion here!

Reply:
I've updated #4291 to match this suggestion:

<img width="735" alt="Screenshot 2024-05-20 at 7 28 45 PM" src="https://github.com/WordPress/openverse/assets/15233243/cd203274-153a-413d-ab4d-a209f25da4f5">

Reply:
This works! I trust @fcoveram's perspective regarding the design approach and what might be best for users here 😄 I think we could find better terminology to use there but that can be discussed in the issues/PRs!